cql2pgjson/src/main/java/org/folio/cql2pgjson/CQL2PgJSON.java

```diff
@@ -491,6 +507,120 @@ private String pg(CQLTermNode node) throws QueryValidationException {
+  private Table checkForForeignLocationOfTerm(CQLTermNode node) {
+    String[] termParts = node.getTerm().split("\\.");
+    //if the table is not supplied we do not have enough information to proceed, thereby assume it is in current table
+    if(termParts.length <= 1) {
+      return null;
+    }
+    return getForeignTable(termParts[0]);
+  }
+
+  //method to determine if the right side of the node is a foreign term and returns the table it is attached to is so
+  private Table checkForForeignLocationOfIndex(CQLTermNode node) {
+    String[] idxParts = node.getIndex().split("\\.");
+    //if the table is not supplied we do not have enough information to proceed, thereby assume it is in current table
+    if(idxParts.length <= 1) {
+      return null;
+    }
+    return getForeignTable(idxParts[0]);
+  }
+
+  private Table getForeignTable(String tableName) {
+    for(Table table : dbSchema.getTables()) {
+      if(table.getTableName().equals(tableName) && !table.getTableName().equals(dbTable.getTableName()) ) {
+        return table;
+      }
+    }
+    return null;
+  }
+
+  private String subQuery(String index,CQLTermNode node, Table correlation) throws QueryValidationException {
+    String[] idxParts = index.split("\\.");
+    String[] termParts = node.getTerm().split("\\.");
+
+    String [] foreignTarget ;
+    CqlModifiers modifiers = new CqlModifiers(node);
+    if (idxParts.length > termParts.length) {
+
+      foreignTarget = idxParts;
+    } else {
+      foreignTarget = termParts;
+    }
+    ForeignKeys fkey = findForeignKey(dbTable.getPkColumnName(),correlation);
+
+    if (fkey == null) {
+      String msg = "subQuery(): No foreignKey for table " + foreignTarget[0] + " found";
+      logger.log(Level.SEVERE, msg);
+      throw new QueryValidationException(msg);
+    }
+    if (fkey.getTargetTable() == null) {
+      String msg = "subQuery: Missing target table for foreignKey field " + fkey.getFieldName();
+      logger.log(Level.SEVERE, msg);
+      throw new QueryValidationException(msg);
+    }
+    try {
+
+      CQL2PgJSON c = new CQL2PgJSON(foreignTarget[0] + ".jsonb");
+      String term = node.getTerm();
+
+      boolean isTermConstant = !tableNamePattern.matcher(term).matches();
+      boolean isTermUUID = uuidPattern.matcher(term).matches();
+
+      String myField = dbTable.getTableName() + ".id";
+      String targetField = index2sqlText(foreignTarget[0] + ".jsonb", fkey.getFieldName());
+      String whereField = index2sqlText(foreignTarget[0] + ".jsonb", foreignTarget[1]);
+      String whereClause = "";
+      String inKeyword = "";
+
+      String indexString = "";
+      String selectString = "";
+      if (isTermConstant) {
+        String termString = "";
+        if(isTermUUID) {
+          termString = "(" + term + ")::UUID";
+          indexString =  "(" + whereField + ")::UUID";
+        } else {
+          termString = wrapInLowerUnaccent("'" + Cql2SqlUtil.cql2like(term) + "'", modifiers);
```

Comment:
This only works for the LIKE operator but not for the = operator.

Reply:
You may want to add a Cql2SqlUtil.cql2quotedLike(String) method.

Reply:
You cannot use modifiers. We dropped supporting CQL modifiers in https://issues.folio.org/browse/CQLPG-81 Note 3.